### PR TITLE
Persistence: replace initialState with emptyState

### LINKED ISSUE #24917: Persistence: initialState => emptyState
Rename `initialState` in persistence to `emptyState` so it explains more that it is the "zero"

### PR DESCRIPTION
**Pull request by richardimaoka**
Closes #24917

### CODE REVIEW

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala [hunk 1]
```diff
@@ -16,7 +16,7 @@ import akka.util.ConstantFun
 @InternalApi
 private[akka] final case class PersistentBehaviorImpl[Command, Event, State](
   persistenceId:     String,
-  initialState:      State,
+  emptyBehavior:     State,
```
Comment (raboof) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala]: `emptyState` here as well I presume?

**Review by raboof — COMMENTED** [review]
One oversight I think, but other than that I think I like the change

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala (hunk 1 above)
Comment (richardimaoka) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala]: Oops, thanks for finding this. Fixed in the new commit.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Review by raboof — APPROVED** [review]

**Review by patriknw — COMMENTED** [review]
thanks

**Comment by notxcain** [comment]
That is very strange decision. `empty` is so narrow compared to `initial`.

**Pull request merged**